program/src/processor.rs

```diff
@@ -35,33 +35,44 @@ impl Processor {
                     return Err(RuleSetError::PayerIsNotSigner.into());
                 }
 
+                // Deserialize RuleSet.
+                let rule_set: RuleSet = rmp_serde::from_slice(&args.serialized_rule_set)
```

Comment:
We want to check that the `rule_set` owner matches the `payer` key so that people can only create rule sets in their own namespace:

```rust
if !mpl_utils::cmp_pubkey(&rule_set.owner, payer_info.key) {
    return Err(RuleSetError::OwnerDoesNotMatchPayer)
}
```

Reply:
FIxed